openedx/core/djangoapps/embargo/views.py

```diff
@@ -2,10 +2,46 @@
 
 from django.http import Http404
 from django.views.generic.base import View
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import permissions
+from rest_framework.response import Response
+from rest_framework.views import APIView
 
 from edxmako.shortcuts import render_to_response
 
 from . import messages
+from .api import check_course_access
+from .permissions import CanCallCheckCourseAccessAPI
+
+
+class CheckCourseAccessView(APIView):
+    permission_classes = (permissions.IsAuthenticated, CanCallCheckCourseAccessAPI)
+
+    def get(self, request, course_ids=[], user=None, ip_address=None):
+        """
+        GET /api/embargo/v1/course_access/
+
+        Arguments:
+            request (HttpRequest)
+
+        Keyword Arguments:
+            course_ids (list): List of course keys strings.
+            user (User): The user that requires course access.
+            ip_address (str): The IP address of the user requiring access.
+
+        Return:
+            Response: True or False depending on the check.
+
+        """
+        course_ids = request.GET.getlist('course_ids', [])
+        user = request.GET.get('user')
+        user_ip_address = request.GET.get('ip_address')
+
+        for course_id in course_ids:
+            if not check_course_access(CourseKey.from_string(course_id), user, user_ip_address):
+                return Response(False)
+        return Response(True)
```

Comment:
Return an object/dict. Expanding on this endpoint in the future will be much easier without making a new version.